src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -229,19 +229,29 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
             {
                 assembly = host.GetEntryPoint(applicationName);
             }
-            // Compilation exceptions either throw FileNotFound or FileLoad exceptions and may change
-            // from version to version. It's safest to catch both types of exceptions
-            catch (Exception ex) when (ex is FileLoadException || ex is FileNotFoundException)
+            catch (Exception ex)
             {
-                if (ex.InnerException is ICompilationException)
+                // Try to find compilation exception recursively to supress its stack
+                var innerException = ex;
+                do
                 {
-                    throw SuppressStackTrace(ex.InnerException);
```

Comment:
The way dnx loads assembly when running on clr\coreclr differs so we get ICompilationException  either wrapped into FileLoadException or first level. So its either loop trough inner exceptions searching for ICompilationException or two if's like:

    if (ex is ICompilationException)
    {
        throw SuppressStackTrace(ex);
    }

    if (ex.InnerException is ICompilationException)
    {
        throw SuppressStackTrace(ex.InnerException);
    }